apps/mitt-konto/src/MittKonto/Main.purs

```diff
@@ -72,37 +68,36 @@ setPayments payments = _ { payments = payments }
 
 data Loading = Loading
 
-type Self = React.Self Props State
-
-component :: React.Component Props
-component = React.createComponent "MittKonto"
-
-app :: Props -> JSX
-app = make component
-  { initialState:
-      { paper: KSF
-      , loggedInUser: Nothing
-      , loading: Nothing
-      , showWelcome: true
-      , alert: Nothing
-      , logger: Sentry.emptyLogger
-      , payments: Nothing
-      , creditCards: []
-      }
-  , didMount
-  , render
+type Self =
+  { state :: State
+  , setState :: (State -> State) -> Effect Unit
   }
 
-didMount :: Self -> Effect Unit
-didMount self = do
+app :: Component {}
+app = do
   sentryDsn <- sentryDsn_
   logger <- Sentry.mkLogger sentryDsn Nothing "mitt-konto"
-  self.setState _ { logger = logger }
+  let initialState =
+        { paper: KSF
+        , loggedInUser: Nothing
+        , loading: Nothing
+        , showWelcome: true
+        , alert: Nothing
+        , payments: Nothing
+        , creditCards: []
+        }
+  component "MittKonto" \_ -> React.do
+    state /\ setState <- useState initialState
+    let self = { state, setState }
+    pure $ render self logger
+
+jsApp :: {} -> JSX
+jsApp = unsafePerformEffect app
 
-render :: Self -> JSX
-render self@{ state, setState } =
+render :: Self -> Sentry.Logger -> JSX
```

Comment:
Is it unavoidable to pass the logger all the time?

Reply:
It's not a real problem, I was just wondering if it was forced by **react**

Reply:
I could have left it just as it was, but my idea was that it's better this way since the only reason we used `setState` to set the logger was because we used `didMount` to set it. No actions done after that point ever changed it. The fewer things we have as state the better.